pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTElement.java

```diff
@@ -29,7 +29,7 @@
 import org.codehaus.groovy.ast.ASTNode;
 import org.jenkinsci.plugins.pipeline.modeldefinition.validator.ModelValidator;
 
-public abstract class ModelASTElement {
+public abstract class ModelASTElement implements Cloneable {
```

Comment:
Why do we want this `Cloneable`?

Reply:
When duplicating stuff later I just cloned it. I was adding thing into the environment and it seemed easier to add it into the base class